spanner/client.go

```diff
@@ -331,6 +335,12 @@ type ClientConfig struct {
 	DirectedReadOptions *sppb.DirectedReadOptions
 
 	OpenTelemetryMeterProvider metric.MeterProvider
+
+	// EnableServerSideTracing indicates whether server side tracing is enabled or not.
```

Comment:
Lets add same comment as that in java PR

Reply:
Makes sense. Updated the opt-in header PR(https://github.com/googleapis/google-cloud-go/pull/10241) as well.

Reply:
Why do we have same changes in 2 different PRs. If https://github.com/googleapis/google-cloud-go/pull/10241/files is valid, then remove these changes from here and lets have the executor changes alone.

Reply:
This PR is created on top of the changes of https://github.com/googleapis/google-cloud-go/pull/10241. Executor changes also includes the usage of `EnableServerSideTracing` flag which is in https://github.com/googleapis/google-cloud-go/pull/10241. That's why copied the changes from that PR.

Considering this PR will be submitted after https://github.com/googleapis/google-cloud-go/pull/10241 is submitted, When we submit this PR, can't we take a pull from main branch?

Reply:
It is better to not have changes from other PR here since it will double the review effort.
Since this PR will any how get submitted after #10241, when you rebase this PR with main you will have those changes applied automatically. However, till then you might see failing builds in this PR which is not a problem.

Lets have executor changes alone in this PR.

Reply:
Removed the opt-in header changes from this PR.